openapi-cli/src/test/resources/ballerina-to-openapi/request_body/union_type.bal

```diff
@@ -0,0 +1,40 @@
+import ballerina/http;
+
+public type ABC record {
+    int id;
+    string name;
+};
+type UnionType ABC|xml;
+
+service /payloadV on new http:Listener(0) {
+    resource function post path(ABC? payload) {
+    }
+    //This needs to be fixed with separated PR
+    resource function post path1(UnionType payload) {
+    }
+    resource function post path2(ABC|xml payload) {
+    }
+    resource function post path3(ABC|http:Remote|xml? payload) {
+    }
+    resource function post path4(@http:Payload ABC|string|xml payload) {
+    }
+    //oneOF- scenarios for application/json
+    resource function post path5(@http:Payload ABC|int payload) {
+    }
+    resource function post path6(@http:Payload int|string payload) {
+    }
+    resource function post path7(@http:Payload map<json>|int[] payload) {
+    }
+    resource function post path8(@http:Payload map<int>|string payload) {
+    }
+    resource function post path9(@http:Payload map<int>|map<string> payload) {
+    }
+
+    // //negative - skip
```

Comment:
Better to remove the commented code.

Reply:
This is added with the purpose of adding negative tests in the future.